src/games/strategy/triplea/ai/proAI/util/ProAttackOptionsUtils.java

```diff
@@ -445,23 +458,29 @@ public void findDefendOptions(final PlayerID player, final List<Territory> myUni
         ProMatches.territoryHasNoEnemyUnitsOrCleared(player, data, clearedTerritories), clearedTerritories, false,
         false);
     findLandMoveOptions(player, myUnitTerritories, moveMap, unitMoveMap, landRoutesMap,
-        Matches.isTerritoryAllied(player, data), new ArrayList<Territory>(), false, false, false);
+        Matches.isTerritoryAllied(player, data), new ArrayList<Territory>(), new ArrayList<Territory>(), false, false,
+        false);
     findAirMoveOptions(player, myUnitTerritories, moveMap, unitMoveMap,
-        ProMatches.territoryIsNotConqueredAlliedLand(player, data), new ArrayList<Territory>(), false, false, false);
+        ProMatches.territoryIsNotConqueredAlliedLand(player, data), new ArrayList<Territory>(),
+        new ArrayList<Territory>(), false, false, false);
     findAmphibMoveOptions(player, myUnitTerritories, moveMap, transportMapList, landRoutesMap,
         Matches.isTerritoryAllied(player, data), new ArrayList<Territory>(), false, false, false);
   }
 
-  public void findMaxEnemyAttackUnits(final PlayerID player, final List<Territory> myConqueredTerritories,
+  public void findEnemyAttackOptions(final PlayerID player, final List<Territory> myConqueredTerritories,
```

Comment:
Agree with both points. In general, I need to use Sets instead of Lists more often. I just tend to overuse Lists because I like their API better and also didn't realize a bunch of them should be Sets. There are also lots of existing 'engine methods' that use Lists so it requires converting somewhere.

I would like to make all the utils classes static. I was going a different direction originally and going to save state in them but most became just pure utility classes. It requires a bunch of refactoring though due to how GameData is shared right now.